app/src/main/java/com/poklad/androidifystore/presentation/ui/screens/auth/forgot_password/ResetPasswordViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.poklad.androidifystore.presentation.ui.screens.auth.forgot_password
+
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.FirebaseUser
+import com.poklad.androidifystore.domain.usecases.ResetPasswordUseCase
+import com.poklad.androidifystore.extensions.log
+import com.poklad.androidifystore.presentation.ui.base.BaseViewModel
+import com.poklad.androidifystore.presentation.ui.screens.auth.utils.AllEvents
+import com.poklad.androidifystore.utils.CoroutineDispatchersProvider
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class ResetPasswordViewModel @Inject constructor(
+    dispatchersProvider: CoroutineDispatchersProvider,
+    private val resetPasswordUseCase: ResetPasswordUseCase
+) : BaseViewModel(dispatchersProvider) {
+    override val coroutineExceptionHandler: CoroutineExceptionHandler
+        get() = CoroutineExceptionHandler { _, throwable ->
+            log(throwable.message.toString())
+        }
+    private val _eventsFlow = MutableSharedFlow<AllEvents>()
```

Comment:
not sure, that it makes sense to store only one flow. It's better to use 1 flow for data and another flow for errors (which you already defined in BaseViewModel)